### PR TITLE
31 - Julia and Melvin - cohort page   student view

### LINKED ISSUE #31: Cohort Page - Student View
Student users who click the "Cohort" link in the left navigation bar should be directed to a page that shows all of the students in their cohort, as well as a list of teachers.

When the user clicks on a student, it should direct them to that student's profile.

Additionally, the student should see high-level statistics about the exercises, units, and modules they have completed, displayed in an "Exercises" section as shown in the design. This section can be hard-coded if exercise functionality has not yet been implemented.

ToDos:

Implement routing to ensure the "Cohort" link directs users to this page.
Display a list of students in the cohort, showing their initials, full name, and an option menu for additional actions.
Display a list of teachers associated with the cohort, including initials, name, and specialization.
Make each student clickable, directing to their profile page upon click.
Add the "My Exercises" section showing modules, units, and exercises completed, which can be hard-coded if necessary.

### PR DESCRIPTION
**Pull request by uthmel**

### CODE REVIEW

**Review by JosteinRuen — DISMISSED** [review]
Amazing work, great functionality. We will look at the styling later 👍

**Review by johnfa1508 — DISMISSED** [review]
This is amazing!

File: src/service/apiClient.js [hunk 1]
```diff
@@ -52,4 +52,4 @@ async function request(method, endpoint, data, auth = true) {
   return response.json();
 }
 
-export { login, getPosts, register, createProfile };
+export { login, getPosts, register, createProfile, get };
```
Comment (dogezen) [inline src/service/apiClient.js:55]: don't expose `get` -> create instead a getUsers function instead in `apiClient.js` and export that function.

File: src/pages/cohort/myCohort.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import jwtDecode from 'jwt-decode';
+
+const MyCohort = () => {
+  const [students, setStudents] = useState([]);
+  const [teachers, setTeachers] = useState([]);
+  const [loggedInUser, setLoggedInUser] = useState([]);
+  const { token } = useAuth();
+
+  useEffect(() => {
+    const loggedInId = jwtDecode(token).userId;
```
Comment (dogezen) [inline src/pages/cohort/myCohort.js:14]: loggedinuser can be obtained from auth context, no need to decode the token

**Review by dogezen — DISMISSED** [review]
1) please update the API call - see feedback
2) ideally, please update getting the logged in user id from auth context and not by parsing token
3) create an issue in backlog to refactor this component so that it can be reused to see any cohort page by cohort id -> this can be ignored for tomorrow, and would be an issue to work on next week, for example, or early tomorrow morning if you wish
4) rename the file to MyCohort.jsx -> although I think the name would be better as `CohortPage`

**Review by allendewilzin — DISMISSED** [review]

**Comment by allendewilzin** [comment]
Ensure you have your branch updated with main.

**Review by allendewilzin — APPROVED** [review]

**Review by johnfa1508 — APPROVED** [review]

**Pull request merged**